src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -346,20 +353,80 @@ public UUID initiateStatelessBdbaScan(BlackDuckRunData blackDuckRunData) throws
     }
 
 
-    private String getScanServicePostEndpoint() {
+    public String getScanServicePostEndpoint() {
         if (detectConfigurationFactory.createScanMode() == BlackduckScanMode.INTELLIGENT) {
             return INTELLIGENT_SCAN_ENDPOINT;
         }
         return RAPID_SCAN_ENDPOINT;
     }
 
-    private String getScanServicePostContentType() {
+    public String getScanServicePostContentType() {
         if (detectConfigurationFactory.createScanMode() == BlackduckScanMode.INTELLIGENT) {
             return INTELLIGENT_SCAN_CONTENT_TYPE;
         }
         return RAPID_SCAN_CONTENT_TYPE;
     }
 
+    public File getContainerScanImage() {
+        Optional<Path> containerImageFilePath = detectConfigurationFactory.getContainerScanFilePath();
+        File containerImageFile = null;
+        if (containerImageFilePath.isPresent()) {
+            containerImageFile = containerImageFilePath.get().toFile();
+        }
+        return containerImageFile;
+    }
+
+    public JSONObject createContainerScanImageMetadata(UUID scanId, NameVersion projectNameVersion) {
+        String scanPersistence = detectConfigurationFactory.createScanMode() == BlackduckScanMode.INTELLIGENT ? "STATEFUL" : "STATELESS";
+        String projectGroupName = detectConfigurationFactory.createProjectGroupOptions().getProjectGroup();
+
+        JSONObject imageMetadataObject = new JSONObject();
+        imageMetadataObject.put("scanId", scanId.toString());
+        imageMetadataObject.put("scanType", "CONTAINER");
+        imageMetadataObject.put("scanPersistence", scanPersistence);
+        imageMetadataObject.put("projectName", projectNameVersion.getName());
+        imageMetadataObject.put("projectVersionName", projectNameVersion.getVersion());
+        imageMetadataObject.put("projectGroupName", projectGroupName);
+
+        return imageMetadataObject;
+    }
+
+    public Response uploadFileToStorageService(BlackDuckRunData blackDuckRunData, String storageServiceEndpoint, File payloadFile, String postContentType)
+        throws IntegrationException {
+        BlackDuckServicesFactory blackDuckServicesFactory = blackDuckRunData.getBlackDuckServicesFactory();
+        BlackDuckApiClient blackDuckApiClient = blackDuckServicesFactory.getBlackDuckApiClient();
+
+        HttpUrl postUrl = new HttpUrl(blackDuckRunData.getBlackDuckServerConfig().getBlackDuckUrl().toString() + storageServiceEndpoint);
+        BlackDuckResponseRequest buildBlackDuckResponseRequest = new BlackDuckRequestBuilder()
+            .postFile(payloadFile, ContentType.create(postContentType))
+            .buildBlackDuckResponseRequest(postUrl);
+
+        try (Response response = blackDuckApiClient.execute(buildBlackDuckResponseRequest)) {
+            return response;
+        } catch (IntegrationException | IOException e) {
```

Comment:
Reason to catch and throw the same exception?

Reply:
Fixed - handled & logged both exceptions separately now. Thanks! (Catch & throw was for more explicit logging/exception messages)